pipelinewise/fastsync/partialsync/mysql_to_snowflake.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+import sys
+import os
+import glob
+
+from argparse import Namespace
+import multiprocessing
+from typing import Union
+from datetime import datetime
+
+from pipelinewise.fastsync.commons.tap_mysql import FastSyncTapMySql
+from pipelinewise.fastsync.commons.target_snowflake import FastSyncTargetSnowflake
+
+from pipelinewise.logger import Logger
+from pipelinewise.fastsync.commons import utils
+from pipelinewise.fastsync.mysql_to_snowflake import REQUIRED_CONFIG_KEYS, tap_type_to_target_type
+from pipelinewise.fastsync.partialsync.utils import load_into_snowflake, upload_to_s3, update_state_file
+
+LOGGER = Logger().get_logger(__name__)
+LOCK = multiprocessing.Lock()
+
+
+def partial_sync_table(args: Namespace) -> Union[bool, str]:
+    """Partial sync table for MySQL to Snowflake"""
+    snowflake = FastSyncTargetSnowflake(args.target, args.transform)
+    tap_id = args.target.get('tap_id')
+
+    try:
+        mysql = FastSyncTapMySql(args.tap, tap_type_to_target_type)
+
+        # Get bookmark - Binlog position or Incremental Key value
+        mysql.open_connections()
+        bookmark = utils.get_bookmark_for_table(args.table, args.properties, mysql)
+        file_parts = _export_source_table_data(args, tap_id, mysql)
+        mysql.close_connections()
+        size_bytes = sum([os.path.getsize(file_part) for file_part in file_parts])
+        s3_keys, s3_key_pattern = upload_to_s3(snowflake, file_parts, args.temp_dir)
+        load_into_snowflake(snowflake, args, s3_keys, s3_key_pattern, size_bytes)
+        update_state_file(args, bookmark, LOCK)
+
+        return True
+    except Exception as exc:
+        LOGGER.exception(exc)
+        return f'{args.table}: {exc}'
+
+
+def _export_source_table_data(args, tap_id, mysql):
+    filename = utils.gen_export_filename(tap_id=tap_id, table=args.table, sync_type='partialsync')
+    filepath = os.path.join(args.temp_dir, filename)
+
+    # Exporting table data
+    where_clause_setting = {
+        'column': args.column,
+        'start_value': args.start_value,
+        'end_value': args.end_value
+    }
+    mysql.copy_table(
+        args.table,
+        filepath,
+        split_large_files=args.target.get('split_large_files'),
+        split_file_chunk_size_mb=args.target.get('split_file_chunk_size_mb'),
+        split_file_max_chunks=args.target.get('split_file_max_chunks'),
+        where_clause_setting=where_clause_setting
+    )
+    file_parts = glob.glob(f'{filepath}*')
+    return file_parts
```

Comment:
I reckon this logic can be a method in the `FastSyncTapMySql` class.

Reply:
- [ ] Move method into the class